### PR TITLE
feat(robot-server, api): async hardware initialization after HTTP server starts.

### LINKED ISSUE #6109: feat(robot-server): starts up immediately without waiting on hardware initialization
The robot-server waits for hardware initialization to complete before starting up. This can take a very long time on a PI or robot. 

The server should be able to accept HTTP requests immediately upon starting while waiting for robot initialization to complete.

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview
The robot-server initializes ThreadManager before starting the HTTP service.  This PR changes that. The ThreadManager is created asynchronously in the Starlette start_up hook. 

A Dependency is used to reject requests that require ThreadManager

# Changelog
* Converted opentrons.initialize to be async
* Create robot_server.HardwareWrapper that create's a task to call `opentrons.initialize`
* Return a 503 when a valid hardware instance is not ready yet
* call systemd.daemon.notify even if using virtual smoothie to support those of us using just a PI 

# Review requests
* is 503 appropriate
* Health will return 503 until hardware is initialized. Is that correct?

# Test plan

## Basic test
- Install on robot. RunApp interaction should be exactly the same is before this change.

## Detailed
- While booting, the run app will not be able to connect. 
- While booting, `curl {robot_ip}:31950/settings` will return settings. 
- While booting, `curl {robot_ip}:31950/health` will return 503 error vs "502 bad gateway"

# Risk assessment
High. Touches most endpoints

closes #6109

┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=535469645) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

**Comment by mcous** [comment]
Pushed this branch to a bare Pi and a OT-2 for initial testing:

- [x] Pi booted successfully and remained up (no systemd timeout)
- [x] OT-2 booted and remained up

File: robot-server/robot_server/main.py [hunk 1]
```diff
@@ -1,64 +1,42 @@
...
+    """Start the uvicorn service."""
     if path:
         log.debug(f"Starting Opentrons application on {path}")
         hostname, port = None, None
     else:
         log.debug(f"Starting Opentrons application on {hostname}:{port}")
         path = None
 
-    fastapi_run(hardware, hostname, port, path)
+    from robot_server.service.app import app
```
Comment (sfoster1) [inline robot-server/robot_server/main.py:22]: Why do we need to import the app here (as opposed to at the top of the file)? Does it enable uvicorn live reloads or something?

File: robot-server/robot_server/main.py [hunk 2]
```diff
@@ -1,64 +1,42 @@
...
-            load_simulator,
-            Path(app_settings.simulator_configuration_file_path))
-    else:
-        # Create the hardware
-        checked_hardware = initialize_api(
-                hardware_server=app_settings.hardware_server_enable,
-                hardware_server_socket=app_settings.hardware_server_socket_path
-        )
+    # TODO Amit 7/14/2020
+    #  Now that aiohttp is goine, we can start the robot-server using uvicorn
```
Comment (sfoster1) [inline robot-server/robot_server/main.py:33]: Could do it in this pr, even!

File: robot-server/robot_server/main.py (hunk 1 above)
Comment (amitlissack) [reply robot-server/robot_server/main.py:22]: Uvicorn does do live reloads.

File: robot-server/robot_server/main.py (hunk 2 above)
Comment (amitlissack) [reply robot-server/robot_server/main.py:33]: OOps. Forgot to update the service file. I'm going to leave this here for backwards compat though.

**Comment by amitlissack** [comment]
> Does look like it'll work, and I really like the other changes around initialization and removing more aiohttp stuff. That said, if the health response is still going to 503 until the hardware is up, I'm not quite sure what deferring initialization actually gets us. The robot still won't appear in the app until the health response gives a 200, I'm pretty sure.

Maybe today it's not that big a deal, but down the road it opens up possibilities:
- Health could be more nuanced: responding 200 with more details about readiness
- Maybe progress notifications of robots connections

**Pull request merged**